### ISSUE TITLE
Provide an identity-map friendly update/remove method

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#2397]
**Is your feature request related to a problem? Please describe.**

Sometimes you have the id of an entity available and want to remove the relationships associated with them, or otherwise update in batch. There are times where some of the items which would match that query, are already loaded in the identify map. Today, the only way to blend the two, would be to run a query to get all of the entities, then update it your self, example:

```typescript
const newCard = new PaymentMethod();

// some logic to populate and persist entity

const customer = await em.findOne(Customer, customerId, {populate: {paymentMethods: true}});

for(const method of customer.paymentMethods){
	method.primary = false;
}

newCard.primary = true;
```

**Describe the solution you'd like**

Due to some of the advanced capabilities of the current QB implementation, it isn't practical to use that interface (eg: you can have a where clause on `expr()`, so we can provide a simpler API, which only supports an `EntityData` like object graph.

```typescript
const customer = await em.update(PaymentMethod, {customer}, {primary: false});
```

The above would effectively do two things:

* Compare the simplified where clause with entities in-memory, and apply the update (`em.assign()`)
* Queue an update query to be executed at flush

**Comment by B4nan** [comment]
Interesting idea, we might wanna update the entity snapshots (originalEntityData) when assigning too, so we do not issue two update queries if there is a match. 

I was actually thinking about having em.insert() too, that would basically partially flush the entities from parameter (cascading their relations, but not fully sure if it should handle more than inserts).

**Comment by parisholley** [comment]
@B4nan is this still a viable idea with v5?

**Comment by B4nan** [comment]
Yes, more like v6 I guess, I already have too many things in the pipeline for 5.x and for v6 I'd like to revisit the EM API a bit, introducing `em.insert/update/delete` that work similarly to the new `em.upsert` - works with entities as well as POJOs, at the same time respects the identity map (if things are already in context, it delegates writing to the flush call). Those would probably replace the `native*` methods.

Speaking of `em.upsert()`, your suggestion is quite aligned with what that method already does.
